azure-pipelines.yml

```diff
@@ -11,26 +11,72 @@ pr:
       - develop
   drafts: false
 
-pool:
-  vmImage: ubuntu-latest
+variables:
+  environmentName: 'team-up-dev'
+  vmImageName: 'ubuntu-latest'
 
-steps:
-  - script: npm ci
-    displayName: 'Install Dependencies'
+stages:
+- stage: Build
+  displayName: Build stage
+  jobs:
+  - job: Build
+    displayName: Building
+    pool:
+      vmImage: $(vmImageName)
 
-  - script: npm run lint:fix
-    displayName: 'Linting'
+    steps:
+      - script: npm ci
+        displayName: 'Installing Dependencies'
 
-  - script: npm test -- --ci --watchAll=false --passWithNoTests
-    displayName: 'Running Unit Test'
+      - script: npm run lint:fix
+        displayName: 'Linting'
 
-  - script: npm run build
-    displayName: 'Build React webApp'
+      - script: npm test -- --ci --watchAll=false --passWithNoTests
+        displayName: 'Running Unit Test'
 
-  - task: PublishBuildArtifacts@1
-    displayName: 'Save Artifact'
-    condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-    inputs:
-      PathtoPublish: '$(Agent.BuildDirectory)/s/build'
-      ArtifactName: 'drop'
-      publishLocation: 'Container'
+      - script: npm run build-prod
+        displayName: 'Building React webApp'
+
+      - task: ArchiveFiles@2
+        displayName: 'Zipping files'
+        inputs:
+          rootFolderOrFile: '$(System.DefaultWorkingDirectory)/build'
+          includeRootFolder: false
+          archiveType: zip
+          archiveFile: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
+          replaceExistingArchive: true
+
+      - upload: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
```

Comment:
these last two tasks, zip and deploy, shouldn't happen for a PR build, please refactor adding conditions

Reply:
yes fixing.
Maybe stage deploy to?

Reply:
add another stage you men? overkill atm, we'd waste precious pipeline seconds in staged handoff